homeassistant/components/remootio/utils.py

```diff
@@ -0,0 +1,114 @@
+"""Utility methods for the Remootio integration."""
+import logging
+from logging import Logger
+
+from aioremootio import ConnectionOptions, LoggerConfiguration, RemootioClient
+from aioremootio.enums import State
+from aioremootio.errors import (
+    RemootioClientAuthenticationError,
+    RemootioClientConnectionEstablishmentError,
+)
+import async_timeout
+
+from homeassistant import core
+from homeassistant.helpers import aiohttp_client
+
+from .const import EXPECTED_MINIMUM_API_VERSION, REMOOTIO_TIMEOUT
+from .exceptions import (
+    CannotConnect,
+    InvalidAuth,
+    UnsupportedRemootioApiVersionError,
+    UnsupportedRemootioDeviceError,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _check_api_version(remootio_client: RemootioClient) -> None:
+    """Check whether the by the given client represented Remootio device uses a supported API version."""
+    api_version: int = await remootio_client.api_version
+    if api_version < EXPECTED_MINIMUM_API_VERSION:
+        raise UnsupportedRemootioApiVersionError(remootio_client, api_version)
+
+
+async def _check_sensor_installed(
+    remootio_client: RemootioClient, raise_error: bool = True
+) -> None:
+    """Check whether the by the given client represented Remootio device has a sensor installed."""
+    if await remootio_client.initialized:
+        if remootio_client.state == State.NO_SENSOR_INSTALLED:
+            if raise_error:
+                raise UnsupportedRemootioDeviceError(remootio_client)
+            else:
+                _LOGGER.error(
+                    f"Your Remootio device isn't supported, possibly because it hasn't a sensor installed. IP [{remootio_client.ip_address}]"
+                )
+
+
+def _handle_exception(ex: Exception) -> None:
+    """Handle errors raised by a Remootio client."""
```

Comment:
Seems like uneccesary function. Just catch the original exceptions directly where needed.

Reply:
I use this method both in the method `utils.get_serial_number` and the method `utils.create_client` to translate exceptions of the library into Home Assistant exceptions.

Reply:
The exceptions you translate into, are mostly example exceptions. They serve no meaning outside your config flow. As such, it is better to propagate the original exception

Reply:
If you think it would be better, I can change it.

Reply:
Fixed.